File: LeetCodeCPlusPlus/py56.cpp
// 合并区间，排序后合并
#include <vector>
using namespace std;
class Solution {
public:
    static bool cmp(const vector<int>&a,const vector<int>&b )
    {
        if(a[0]==b[0])
            return a[1]>b[1];
        return a[0]<b[0];
    }
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        vector<vector<int>> result;
        if (intervals.empty())
            return result;
        sort(intervals.begin(), intervals.end(), cmp);
        int begin = intervals[0][0];
        int end = intervals[0][1];
        for (int i=0;i<intervals.size(); i++) {
            if (intervals[i][0]>end) {
                result.push_back({begin, end});
                begin = intervals[i][0];
                end = intervals[i][1];
            }else if (intervals[i][1]>end) 
                end = intervals[i][1];
        }
        result.push_back({begin,end});
        return result;
    }
};

File: LeetCodePython/py56.py
# 合并区间； 排序后合并
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if not intervals:
            return []
     
        result = []
        intervals.sort()
        begin = intervals[0][0]
        end = intervals[0][1]
        for item_begin, item_end in intervals:
            if item_begin > end:  # 超出当前的边界，不能合并，则加入结果数组 
                result.append([begin, end])
                begin = item_begin
                end = item_end
            elif item_end > end:  # 不超出边界，则判断是否需要更新边界
                end = item_end
        result.append([begin, end]) # 最后一个边界加入
        return result